rfcs/0000-object-field.md

```diff
@@ -0,0 +1,118 @@
+# 0000: Object Fields
+<!--^ The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC, taking care not to conflict with other RFCs.-->
+
+- Stage: **0 (strawperson)** <!-- Update to reflect target stage -->
+- Date: **7/7/2020** <!-- Update to reflect date of most recent stage advancement -->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+Numerous SaaS Cloud Providers provide event data where the event reflects an action (Typically a CRUD action) on what can broadly be considered an "object." For some providers that Object may be a "file" wherein the existing ECS `file` fields are adequate; however other providers (Workday, Salesforce, Service Now, Zoom, etc.) an object can represent some other concept like a "record" or "setting" or "meeting." There presently isn't any location within ECS to document these as there is with "file," forcing any representation to be pushed into a custom field. This RFC proposes a new top-level field called `object` that will represent these instances.
+
+## Fields
+
+<!--
+Stage: 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
+-->
+This RFC calls for the introduction of a top-level `object` field with an initial fieldset of the following child fields:
+
+| field | type | description |
+| --- | --- | --- |
+| `object.name` | keyword | Name of the Object |
+| `object.id` | keyword | ID of the Object |
+| `object.type` | keyword | Type of Object Represented (e.g. `record`, `meeting`, `repository`, `organization`, etc.) |
+| `object.additional_details` | object | Custom Key/Value pairs representing data relevant to the object |
+| `object.owner.id` | keyword | ID of the Object Owner |
```

Comment:
:grey_question: How do these `owner` fields overlap/co-exist with https://www.elastic.co/guide/en/ecs/current/ecs-user.html ?

Reply:
It's very likely that owner could just be a nested user object; I haven't encountered an instance yet where that wouldn't work.

Reply:
> In general user.* at the root of the event is the "doer". If we need to capture an object's owner, we could nest the user fields in an additional location. 

@webmat That makes a lot of sense, but it's hard to discern the "doer" inclination from the description of `user.*` which reads "The user fields describe information about the user that is relevant to the event." The spec might benefit from making that distinction more explicit to the reader.

Reply:
Agreed. We'll be improving on this soon.